datasets/ky/legislativeassembly/crawler.py

```diff
@@ -0,0 +1,128 @@
+
+    # Find the indexes in the text were each position is written
+    positions_idx = []
+    for pos in positions:
+        positions_idx += [(pos, page_text.find(pos))]
+
+    def get_positions(s):
+        """
+        Find information about positions from a string
+        """
+        positions_init = ["Minister", "Member", "Premier", "Deputy"]
+        p = []
+        new_pos = []
+
+        # Split text in words
+        words = s.split(" ")
+
+        advance = False
+        for i, w in enumerate(words):
+            # We need to advance when the position is Deputy, because
+            # the next word is Premier and the position is 'Deputy Premier'
+            # and not 'Deputy' and 'Premier'
+            if advance:
+                advance = False
+                new_pos += [w]
+                continue
+            if w == "Deputy":
+                advance = True
+
+            # We get all the subsequent words until we find a new word
+            # representing a position
+            if i > 0 and w in positions_init or i == len(words) - 1:
+                # We found another position, save what we have until now
+                # and start getting a new position. Remove ' and' from the
+                # position if it exists.
+                if i == len(words) - 1:
+                    new_pos += [w]
+                p += [" ".join(new_pos).removesuffix(" and").strip()]
+                new_pos = [w]
+            else:
+                # Add word to position description
+                new_pos += [w]
+        return p
+
+    persons = []
+    for i in range(len(positions_idx) - 1):
+        if positions_idx[i][0] != "Cabinet Ministers":
+            parts = (
+                page_text[positions_idx[i][1] : positions_idx[i + 1][1]]
+                .split("Hon.")[1]
+                .split(",")
+            )
+            new_person = {}
+            new_person["name"] = parts[0].strip()
+            positions = parts[1:]
+            new_person["positions"] = [
+                get_positions(p.replace("MP", "").strip())
+                for p in positions
+                if "MP" in p
+            ][0]
+            persons += [new_person]
+        else:
+            c = page_text[positions_idx[i][1] : positions_idx[i + 1][1]]
+            for cm in c.split("Hon.")[1:]:
+                cm_parts = cm.split("MP,")
+                new_person = {}
+                new_person["name"] = cm_parts[0].split(",")[0].strip()
+                new_person["positions"] = [
+                    p.strip() for p in "".join(cm_parts[1:]).strip().split(" and ")
+                ]
+
+                persons += [new_person]
+
+    for person in persons:
+        person_proxy = context.make("Person")
+        h.apply_name(person_proxy, full=person["name"])
+        # TODO: Make a better ID
+        person_proxy.id = "legislativeassembly-{}".format(
+            person["name"].replace(" ", "_")
+        )
```

Comment:
You could use https://zavod.opensanctions.org/context/#zavod.context.Context.make_id which should automatically generate `ky-la`-prefixed valid IDs for you.

These warnings are due to capitals in the current IDs 
```
2023-09-04 12:38:09 [warning  ] Rejected property value        [zavod.entity] dataset=ky_legislativeassembly entity_id=ky-la-fa6e409ac684162830bbc491b591a1666feeaef3 prop=holder value=legislativeassembly-Wayne_Panton
```